src/mon/OSDMonitor.cc

```diff
@@ -2917,6 +2917,48 @@ bool OSDMonitor::preprocess_command(MonOpRequestRef op)
     }
     rdata.append(ds);
   }
+  else if (prefix == "osd pending_pg_removal") {
+    int32_t osdid = -1;
+    cmd_getval(g_ceph_context, cmdmap, "id", osdid);
+    ceph::unordered_map<int32_t, osd_stat_t> osd_stat_map = mon->pgmon()->pg_map.osd_stat;
+    epoch_t epoch_recent_pool_delete = get_epoch_recent_pool_delete();
+    if (osdid != -1) {
+      osd_stat_t osd_stat = osd_stat_map[osdid];
+      if ((osd_stat.pending_pg_removal > 0) ||  
+          (osd_stat.cur_epoch < epoch_recent_pool_delete)) {
+        ds << " osd." << osdid << " has " << osd_stat.pending_pg_removal << " pgs to be removed" \
+           << " or osdmap epoch " << osd_stat.cur_epoch << " on osd." << osdid \
+           << " is lower than " << epoch_recent_pool_delete  \
+           << " that a pool removal happended on monitor \n";
+      } else {
+        ds << " osd." << osdid << " has no pending pgs to be removed" \
+           << " and osdmap epoch " << osd_stat.cur_epoch << " on osd." << osdid \
+           << " is greater than " << epoch_recent_pool_delete  \
+           << " that a pool removal happended on monitor \n";
+      }
+    } else {
+      bool pending = false;
+      for (ceph::unordered_map<int32_t, osd_stat_t>::iterator it = osd_stat_map.begin(); \
+        it != osd_stat_map.end(); it++ ) {
+        if ((it->second.pending_pg_removal > 0) ||
```

Comment:
Do you really need this part that checks all OSDs in a single command?

If not, I'd prefer to avoid extending the data structures and just query individual OSDs via 'ceph tell osd.NNN ...'.  IIRC there is already a command (status?) that includes the osd epoch lower bound.

Actually, having that lower bound value in the osd_stat_t seems useful. But the pgs pending removal seems a bit excessive.. can you describe your use case?